File: ruby/1/06.rb
module Util
  def ngram(n)
    self.each_cons(n).map(&:join)
  end
end

class Array
  include Util
end

str1 = "paraparaparadise"
str2 = "paragraph"

X = str1.chars.ngram(2)
Y = str2.chars.ngram(2)

p X | Y
p X & Y
p X - Y
puts X.include?("se")
puts Y.include?("se")

File: python/1/06.py
def ngram(s, n):
    return [s[i:i+n] for i in range(len(s)-1)]

s1 = "paraparaparadise"
s2 = "paragraph"

X = ngram(s1, 2)
Y = ngram(s2, 2)

for item in sorted(set(X) | set(Y)):
    print(item)
print()

for item in sorted(set(X) & set(Y)):
    print(item)
print()

for item in sorted(set(X) - set(Y)):
    print(item)
print()

print('se' in X)
print('se' in Y)
